components/layout_thread/lib.rs

```diff
@@ -1066,6 +1080,9 @@ impl LayoutThread {
     fn handle_reflow<'a, 'b>(&mut self,
                              data: &mut ScriptReflowResult,
                              possibly_locked_rw_data: &mut RwData<'a, 'b>) {
+        // Set layout query timestamp
+        data.reflow_goal.set_timestamp();
```

Comment:
This overwrites the timestamp that was stored when the query was created with the current time when we start processing it. Instead, we want to measure the difference between the current time and when the query was created.